User supplied room ids

vivangkumar opened the pull request:
### What?

Allows providing ids to rooms.

### Why?


----

- [x] CHANGELOG updated if relevant?

vivangkumar committed:
User supplied room ids

- Move to v6 api
- deleteMessage now accepts a room_id and id has been renamed to message_id

callum-oakley reviewed (approved):
haha that's better :)

vivangkumar committed:
Fix tests

Comparing array elements to hardcoded positions is going to result in
lots of failures since order isn't guaranteed.
Instead introduce a helper function that checks if any of the
elements in the array has a matching value for a given key
